### PR TITLE
Use SetPlayerName instead of cl_name in CTFBotSpawner::Spawn

### PR DESCRIPTION
**Pull request by Thestickman391**

### CODE REVIEW

**Comment by maximsmol** [comment]
Does this come with any kind of improvement? Or is this just because it makes more sense?

**Comment by melvyn2** [comment]
It would allow bots to work properly even if the `cl_name` command is blacklisted, which IMO it should be.

File: game/server/tf/player_vs_environment/tf_populator_spawners.cpp [hunk 1]
```diff
@@ -1091,7 +1091,7 @@ bool CTFBotSpawner::Spawn( const Vector &rawHere, EntityHandleVector_t *result )
 		}
 
 		// set name
-		engine->SetFakeClientConVarValue( newBot->edict(), "name", m_name.IsEmpty() ? "TFBot" : m_name.Get() );
+		newBot->SetPlayerName(m_name.Get());
```
Comment (maximsmol) [inline game/server/tf/player_vs_environment/tf_populator_spawners.cpp:1094]: Add the `IsEmpty` check from the old version.

**Review by maximsmol — CHANGES_REQUESTED** [review]

**Comment by Thestickman391** [comment]
Made requested changes

**Comment by mastercoms** [comment]
What sort of new pathways are we going through when we use SetPlayerName?

**Pull request closed without merging**